--- scintilla/gtk/makefile.orig	2011-05-02 11:11:17.000000000 +0200
+++ scintilla/gtk/makefile	2011-06-23 10:07:29.202630717 +0200
@@ -6,7 +6,7 @@
 # Builds for GTK+ 2 and no longer supports GTK+ 1.
 # Also works with ming32-make on Windows.
 
-.SUFFIXES: .cxx .c .o .h .a
+.SUFFIXES: .cxx .c .o .h .a .lo .la
 ifdef CLANG
 CC = clang
 CCOMP = clang
@@ -36,7 +36,7 @@
 COMPLIB=..\bin\scintilla.a
 else
 DEL = rm -f
-COMPLIB=../bin/scintilla.a
+COMPLIB=../bin/libscintilla.la
 endif
 
 vpath %.h ../src ../include ../lexlib
@@ -55,22 +55,27 @@
 endif
 
 ifdef DEBUG
-CXXFLAGS=-DDEBUG -g $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS=-DDEBUG $(OPTFLAGS) $(CXXBASEFLAGS) $(THREADFLAGS)
 else
-CXXFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS=-DNDEBUG $(OPTFLAGS) $(CXXBASEFLAGS) $(THREADFLAGS)
 endif
 
 CFLAGS:=$(CXXFLAGS)
 
 CONFIGFLAGS:=$(shell pkg-config --cflags $(GTKVERSION))
-MARSHALLER=scintilla-marshal.o
+MARSHALLER=scintilla-marshal.lo
 
 .cxx.o:
 	$(CC) $(CONFIGFLAGS) $(CXXFLAGS) -c $<
 .c.o:
 	$(CCOMP) $(CONFIGFLAGS) $(CFLAGS) -w -c $<
 
-LEXOBJS:=$(addsuffix .o,$(basename $(notdir $(wildcard ../lexers/Lex*.cxx))))
+.cxx.lo:
+	libtool --tag=CXX --mode=compile $(CC) $(CONFIGFLAGS) $(CXXFLAGS) -c $<
+.c.lo:
+	libtool --tag=CC --mode=compile $(CCOMP) $(CONFIGFLAGS) $(CXXFLAGS) -w -c $<
+
+LEXOBJS:=$(addsuffix .lo,$(basename $(notdir $(wildcard ../lexers/Lex*.cxx))))
 
 all: $(COMPLIB)
 
@@ -80,14 +85,17 @@
 deps:
 	$(CC) -MM $(CONFIGFLAGS) $(CXXFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak
 
-$(COMPLIB): Accessor.o CharacterSet.o LexerBase.o LexerModule.o LexerSimple.o StyleContext.o WordList.o \
-	CharClassify.o Decoration.o Document.o PerLine.o Catalogue.o CallTip.o \
-	ScintillaBase.o ContractionState.o Editor.o ExternalLexer.o PropSetSimple.o PlatGTK.o \
-	KeyMap.o LineMarker.o PositionCache.o ScintillaGTK.o CellBuffer.o ViewStyle.o \
-	RESearch.o RunStyles.o Selection.o Style.o Indicator.o AutoComplete.o UniConversion.o XPM.o \
+$(COMPLIB): Accessor.lo CharacterSet.lo LexerBase.lo LexerModule.lo LexerSimple.lo StyleContext.lo WordList.lo \
+	CharClassify.lo Decoration.lo Document.lo PerLine.lo Catalogue.lo CallTip.lo \
+	ScintillaBase.lo ContractionState.lo Editor.lo ExternalLexer.lo PropSetSimple.lo PlatGTK.lo \
+	KeyMap.lo LineMarker.lo PositionCache.lo ScintillaGTK.lo CellBuffer.lo ViewStyle.lo \
+	RESearch.lo RunStyles.lo Selection.lo Style.lo Indicator.lo AutoComplete.lo UniConversion.lo XPM.lo \
 	$(MARSHALLER) $(LEXOBJS)
-	$(AR) rc $@ $^
-	$(RANLIB) $@
+	libtool --tag=CXX --mode=link $(CC) -o $@ $^ -rpath $(libdir) $(shell pkg-config --libs $(GTKVERSION))
+
+install: $(COMPLIB)
+	install -d $(DESTDIR)$(libdir)
+	libtool --mode=install install $(COMPLIB) $(DESTDIR)$(libdir)
 
 # Automatically generate header dependencies with "make deps"
 include deps.mak
