--- scintilla/gtk/makefile.orig	2019-04-10 07:15:44.649988285 +0200
+++ scintilla/gtk/makefile	2019-04-10 09:41:25.092637245 +0200
@@ -6,7 +6,7 @@
 # Builds for GTK+ 2 and no longer supports GTK+ 1.
 # Also works with ming32-make on Windows.
 
-.SUFFIXES: .cxx .c .o .h .a .list
+.SUFFIXES: .cxx .c .o .h .a .lo .la .list
 
 srcdir ?= .
 
@@ -49,7 +49,7 @@
 COMPLIB=$(srcdir)\..\bin\scintilla.a
 else
 DEL = rm -f
-COMPLIB=$(srcdir)/../bin/scintilla.a
+COMPLIB=$(srcdir)/../bin/scintilla.la
 endif
 
 vpath %.h $(srcdir) $(srcdir)/../src $(srcdir)/../include $(srcdir)/../lexlib
@@ -74,18 +74,18 @@
 
 ifdef DEBUG
 ifdef CLANG
-CTFLAGS=-DDEBUG -g -fsanitize=$(SANITIZE) $(CXXBASEFLAGS) $(THREADFLAGS)
+CTFLAGS=-DDEBUG $(OPTFLAGS) -fsanitize=$(SANITIZE) $(CXXBASEFLAGS) $(THREADFLAGS)
 else
-CTFLAGS=-DDEBUG -g $(CXXBASEFLAGS) $(THREADFLAGS)
+CTFLAGS=-DDEBUG $(OPTFLAGS) $(CXXBASEFLAGS) $(THREADFLAGS)
 endif
 else
-CTFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS) $(THREADFLAGS)
+CTFLAGS=-DNDEBUG $(OPTFLAGS) $(CXXBASEFLAGS) $(THREADFLAGS)
 endif
 
 CXXTFLAGS:=--std=gnu++17 $(CTFLAGS) $(REFLAGS)
 
 CONFIGFLAGS:=$(shell pkg-config --cflags $(GTKVERSION))
-MARSHALLER=scintilla-marshal.o
+MARSHALLER=scintilla-marshal.lo
 
 all: $(COMPLIB)
 
@@ -97,6 +97,11 @@
 .c.o:
 	$(CC) $(CONFIGFLAGS) $(CTFLAGS) $(CFLAGS) -w -c $<
 
+.cxx.lo:
+	libtool --tag=CXX --mode=compile $(CXX) $(CONFIGFLAGS) $(CXXTFLAGS) $(CXXFLAGS) -c $<
+.c.lo:
+	libtool --tag=CC --mode=compile $(CC) $(CONFIGFLAGS) $(CTFLAGS) $(CFLAGS) -w -c $<
+
 GLIB_GENMARSHAL = glib-genmarshal
 GLIB_GENMARSHAL_FLAGS = --prefix=scintilla_marshal
 
@@ -105,7 +110,7 @@
 .list.c:
 	$(GLIB_GENMARSHAL) --body $(GLIB_GENMARSHAL_FLAGS) $< > $@
 
-LEXOBJS:=$(addsuffix .o,$(basename $(sort $(notdir $(wildcard $(srcdir)/../lexers/Lex*.cxx)))))
+LEXOBJS:=$(addsuffix .lo,$(basename $(sort $(notdir $(wildcard $(srcdir)/../lexers/Lex*.cxx)))))
 
 analyze:
 	clang --analyze $(CONFIGFLAGS) $(CXXTFLAGS) $(CXXFLAGS) $(srcdir)/*.cxx $(srcdir)/../src/*.cxx $(srcdir)/../lexlib/*.cxx $(srcdir)/../lexers/*.cxx
@@ -117,15 +122,18 @@
 deps.mak:
 	$(CXX) -MM $(CONFIGFLAGS) $(CXXTFLAGS) $(DEPSRCS) | sed -e 's/\/usr.* //' | grep [a-zA-Z] > $@
 
-$(COMPLIB): Accessor.o CharacterSet.o DefaultLexer.o LexerBase.o LexerModule.o LexerSimple.o StyleContext.o WordList.o \
-	CharClassify.o Decoration.o Document.o PerLine.o Catalogue.o CallTip.o CaseConvert.o CaseFolder.o \
-	ScintillaBase.o ContractionState.o EditModel.o Editor.o EditView.o ExternalLexer.o MarginView.o \
-	PropSetSimple.o PlatGTK.o \
-	KeyMap.o LineMarker.o PositionCache.o ScintillaGTK.o ScintillaGTKAccessible.o CellBuffer.o CharacterCategory.o ViewStyle.o \
-	RESearch.o RunStyles.o Selection.o Style.o Indicator.o AutoComplete.o UniConversion.o XPM.o \
+$(COMPLIB): Accessor.lo CharacterSet.lo DefaultLexer.lo LexerBase.lo LexerModule.lo LexerSimple.lo StyleContext.lo WordList.lo \
+	CharClassify.lo Decoration.lo Document.lo PerLine.lo Catalogue.lo CallTip.lo CaseConvert.lo CaseFolder.lo \
+	ScintillaBase.lo ContractionState.lo EditModel.lo Editor.lo EditView.lo ExternalLexer.lo MarginView.lo \
+	PropSetSimple.lo PlatGTK.lo \
+	KeyMap.lo LineMarker.lo PositionCache.lo ScintillaGTK.lo ScintillaGTKAccessible.lo CellBuffer.lo CharacterCategory.lo ViewStyle.lo \
+	RESearch.lo RunStyles.lo Selection.lo Style.lo Indicator.lo AutoComplete.lo UniConversion.lo XPM.lo \
 	$(MARSHALLER) $(LEXOBJS)
-	$(AR) $(ARFLAGS) $@ $^
-	$(RANLIB) $@
+	libtool --tag=CXX --mode=link $(CXX) -o $@ $^ -rpath $(libdir) -version-info 4:0:0 $(shell pkg-config --libs $(GTKVERSION))
 
+install: $(COMPLIB)
+	install -d $(DESTDIR)$(libdir)
+	libtool --mode=install install $(COMPLIB) $(DESTDIR)$(libdir)
+ 
 # Automatically generate header dependencies with "make deps"
 include deps.mak
